http-gateway/web/src/containers/things/hooks.js

```diff
@@ -59,7 +59,20 @@ export const useThingDetails = deviceId => {
         let updatedLinks = []
 
         if (event === resourceEventTypes.ADDED) {
-          updatedLinks = data.links.concat(resource)
+          const linkExists =
+            data.links.findIndex(link => link.href === resource.href) !== -1

```

Comment:
maybe use some()

Reply:
or maybe just do the map, or reduce.  You seem to be doing the same comparison twice.

Reply:
Yes, because once I check if the resource is in the array, if it is, I will have to update it, if it isn't, I need to update it. I cannot achieve this with map nor reduce, since I would be only looping through the resource list I already have.